### DISCUSSION TITLE
need to install numba separately?

### DISCUSSION THREAD

**Discussion by stathisraptis** [alan-turing-institute/sktime#4380] [Q&A]
Hello, 

I wanted to run HC2 for a multivariable classification problem and it showed I need to install numba.
I went ahead and reinstalled sktime with all dependencies (pip install sktime[all_extras]) succesfully, but it showed the same message again. 
Isn't numba supposed to be installed with all extras?

Thanks

**Comment by stathisraptis** [comment]
I now saw that numba is not supported in python 3.11 and that is probably the issue. If that's the case, shouldn't it be noted that not all features are supported in 3.11?

**Reply by fkiraly** [reply]
Yes, exactly. - some estimators require dependencies that are not compatible with python 3.11 (yet). In particular, all estimators that require `numba` are not available on 3.11, until `numba` becomes available on python 3.11.

`sktime` can still be installed on 3.11 though, without `numba`.

**Reply by fkiraly** [reply]
from an architectural perspective, `sktime` (with minimal dependencies) is the general framework. Some estimators may require additional package dependencies, you can see them by inspecting the `python_dependencies` tag. Some estimators may require specific versions of python, you can see these via the `python_version` tag. The python version may be restricted beyond the `python_version` tag if a dependency is not available on a python version, `numba` would be an example (not 3.11 compatible).
